src/App.jsx

```diff
@@ -1,10 +1,55 @@
-import React from 'react';
-import './App.css';
+import { React, useState } from 'react';
+import axios from 'axios';
+
+const geocodingURI = 'http://api.openweathermap.org/geo/1.0/direct?q=';
+const weatherApiKey = process.env.REACT_APP_WEATHER_API_KEY;
 
 function App() {
+  const [geoData, setGeoData] = useState({});
+
+  const retrieveGeoData = async (city = '') => {
+    try {
+      const response = await axios.get(
+        `${geocodingURI}${city}&appid=${weatherApiKey}`
+      );
+
+      const { data } = response;
+      const [location] = data;
+      const coordinates = {
+        longitude: location.lon,
+        latitude: location.lat,
+      };
+
+      setGeoData(coordinates);
```

Comment:
I think I'm getting a small issue where the first call to `retrieveGeoData` returns an empty object.  

Subsequent calls seem to work as intended.

Reply:
Are you console logging geoData right after the` setGeoData()` call?

Reply:
Are you getting that error for the current, hourly, and daily weather api?

Reply:
Yeah I'm just using the existing console logs.  

Current, hourly, and daily don't appear to replicate the same issue, but they also aren't taking direct input.

Reply:
What do you mean by direct input?

Reply:
Oh I meant search input passed to the function.

Reply:
Ahh okay I see, yeah if you console log directly after like I did, you won't see anything b/c the set function is async by default. But I believe React does update it when it is passed as a prop or used in a context. I'll look into it a bit more.

Reply:
Okay, I think you're right it has something to do with the set function. We're you also getting some inconsistent data?

![image](https://github.com/awong-pdx/frontend-weather-dashboard/assets/74124296/dc59dfc6-1fa3-41e0-be01-651a42733aa1)

Reply:
Yeah I forgot mention, since it doesn't update right away, when you enter a city name and console.log the result, it shows the result of the previous city lat/lon.